flink-connector-bigquery-common/src/main/java/com/google/cloud/flink/bigquery/common/utils/AvroToBigQuerySchemaTransform.java

```diff
@@ -0,0 +1,456 @@
+         * AvroSchema Logical Type to StandardSQLTypeName.
+         * LOGICAL_AVRO_TYPES_TO_BQ_TYPES: Map containing mapping from Logical Avro Schema Type to StandardSQLTypeName.
+         */
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES = new HashMap<>();
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(LogicalTypes.date().getName(), StandardSQLTypeName.DATE);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timestampMicros().getName(), StandardSQLTypeName.TIMESTAMP);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timestampMillis().getName(), StandardSQLTypeName.TIMESTAMP);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.uuid().getName(), StandardSQLTypeName.STRING);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timeMillis().getName(), StandardSQLTypeName.TIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.timeMicros().getName(), StandardSQLTypeName.TIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.localTimestampMillis().getName(), StandardSQLTypeName.DATETIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put(
+                LogicalTypes.localTimestampMicros().getName(), StandardSQLTypeName.DATETIME);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put("geography_wkt", StandardSQLTypeName.GEOGRAPHY);
+        LOGICAL_AVRO_TYPES_TO_BQ_TYPES.put("Json", StandardSQLTypeName.JSON);
+    }
+
+    /**
+     * Converts an Avro schema to a BigQuery schema.
+     *
+     * <p>This method transforms an Avro {@link org.apache.avro.Schema} into a BigQuery {@link
+     * Schema}. It iterates through the fields of the Avro schema, converts each field to its
+     * BigQuery equivalent, and constructs a BigQuery schema with the resulting fields.
+     *
+     * <p>For each Avro field, the method extracts the field's name and documentation (if
+     * available), and uses this information to build the corresponding BigQuery field.
+     *
+     * <p>The Avro schema must be of type Avro RECORD
+     *
+     * @param avroSchema The Avro schema to convert.
+     * @return The converted BigQuery {@link Schema}.
+     */
+    public static Schema getBigQuerySchema(org.apache.avro.Schema avroSchema) {
+        Preconditions.checkNotNull(avroSchema, "The avro schema of the record cannot be null.");
+        Preconditions.checkNotNull(
+                avroSchema.getType(), "The avro schema of the record must have a type.");
+        if (avroSchema.getType() != org.apache.avro.Schema.Type.RECORD) {
+            throw new IllegalArgumentException(
+                    "The Avro Schema must be of RECORD type to convert to BigQuery columns. Found Schema: "
+                            + avroSchema);
+        }
+        Preconditions.checkState(
+                !avroSchema.getFields().isEmpty(),
+                "The avro record schema must have at least one field.");
```

Comment:
Instead of preconditions, prefer throwing explicit Exceptions such as IllegalArgumentException.

Reply:
This is applicable to rest of the file

Reply:
Done. Removed Preconditions usage.